.github/workflows/update-getting-started-guide.yml

```diff
@@ -0,0 +1,74 @@
+name: Update getting started guide
+
+on:
+  release:
+    types: ['released']
+  pull_request:
+    branches: ['main', 'release/**']
+    paths:
+      # Run workflow on changes to the workflow definition itself to spot check
+      # the core release version update functionality.
+      - '.github/workflows/update-getting-started-guide.yml'
+      # Run workflow on changes to the getting started guide to verify if any
+      # changes to the documentation do not break the update automation.
+      - 'docs/getting-started.md'
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-version:
+    runs-on: ubuntu-20.04
+    env:
+      RELEASE_TAG: ''
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Mock release tag on pull request
+        if: github.event_name == 'pull_request'
+        run: echo "RELEASE_TAG=v0.0.0-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+
+      - name: Set published release tag
+        if: github.event_name == 'release'
+        run: echo "RELEASE_TAG=${{ github.event.release.tag_name }}" >> $GITHUB_ENV
+
+      - name: Update getting started version
+        run: |
+          release_tag=${{ env.RELEASE_TAG }}
+          release_version=${release_tag/v/}
+          sed -i -E "s/version=\"([0-9]+\.){2}[0-9]+\"/version=\"${release_version}\"/" docs/getting-started.md
+
+      - name: Assert version was updated correctly
+        if: github.event_name == 'pull_request'
+        run: |
+          release_tag=${{ env.RELEASE_TAG }}
+          release_version=${release_tag/v/}
+          output=$(git diff --exit-code) && {
```

Comment:
Can we move this to a script?

I think it's easier to discover, tests, and understand if the github action yaml only interacts with github env and then calls a script.

Reply:
Sure. Normally we through scripts into `scripts/`; in this case the script is an assertion to validate GitHub Actions changes so it might not be useful into many more contexts. Would `.github/scripts` be more appropriate here?